src/gluonts/dataset/rolling_dataset.py

```diff
@@ -0,0 +1,174 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from typing import Optional
+
+import numpy as np
+import pandas as pd
+from gluonts.dataset.common import Dataset
+from gluonts.dataset.util import to_pandas
+from pydantic import BaseModel
+
+
+class StepStrategy(BaseModel):
+    """
+    Removes datapoints equivalent to step_size for each iteration until
+    amount of data left is less than prediction_length
+
+    Parameters
+    ----------
+    prediction_length
+        The prediction length of the Predictor that the dataset will be
+        used with
```

Comment:
description of `step_size` is missing here

Reply:
Added it